tests/key_field/key_field_sample_generated.h

```diff
@@ -170,13 +200,362 @@ inline bool operator!=(const Bar &lhs, const Bar &rhs) {
 }
 
 
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) Color FLATBUFFERS_FINAL_CLASS {
+ private:
+  float rgb_[3];
+  uint8_t tag_;
+  int8_t padding0__;  int16_t padding1__;
+
+ public:
+  static const flatbuffers::TypeTable *MiniReflectTypeTable() {
+    return ColorTypeTable();
+  }
+  Color()
+      : rgb_(),
+        tag_(0),
+        padding0__(0),
+        padding1__(0) {
+    (void)padding0__;
+    (void)padding1__;
+  }
+  Color(uint8_t _tag)
+      : rgb_(),
+        tag_(flatbuffers::EndianScalar(_tag)),
+        padding0__(0),
+        padding1__(0) {
+    (void)padding0__;
+    (void)padding1__;
+  }
+  Color(flatbuffers::span<const float, 3> _rgb, uint8_t _tag)
+      : tag_(flatbuffers::EndianScalar(_tag)),
+        padding0__(0),
+        padding1__(0) {
+    flatbuffers::CastToArray(rgb_).CopyFromSpan(_rgb);
+    (void)padding0__;
+    (void)padding1__;
+  }
+  const flatbuffers::Array<float, 3> *rgb() const {
+    return &flatbuffers::CastToArray(rgb_);
+  }
+  flatbuffers::Array<float, 3> *mutable_rgb() {
+    return &flatbuffers::CastToArray(rgb_);
+  }
+  bool KeyCompareLessThan(const Color * const o) const {
+    return KeyCompareWithValue(o->rgb()) < 0;
+  }
+  int KeyCompareWithValue(const flatbuffers::Array<float, 3> *_rgb) const {
+    const flatbuffers::Array<float, 3> *curr_rgb = rgb();
+    for (flatbuffers::uoffset_t i = 0; i < curr_rgb->size(); i++) {
```

Comment:
I wonder if we should just make a generic comparer for arrays in the `flatbuffers::Array` class/file, then the generated code doesn't have to be as bloated, it could just do `return CompareArrays(rgb(), *_rgb);`

Reply:
But I don't think generic comparator for arrays can handle the case of array of structs.

Reply:
It should handle the bulk of the cases for native types that have `==` and `<` and `>` defined right? Then for generated structs, we can just emit overloads for those operators and it should all work.

Reply:
Another problem with using `==` `<`  `>` is, we maybe need to go through the array twice to get the result we want. For example if we have array1 < array2. In `KeyCompareWithValue`, if I first check `array1==array2` and then 'array1 < array2', we'll end up traverse the array twice, so it's gonna be slower. What do you think?